client/my-sites/plugins/notices.jsx

```diff
@@ -3,160 +3,153 @@
+			this.showNotification();
+		}
+	}
 
-	const numberOfSites = uniqBy( filteredLogs, 'site.ID' ).length;
-	const numberOfPlugins = uniqBy( filteredLogs, 'plugin.slug' ).length;
+	extractNoticeProps( { completedNotices, errorNotices, inProgressNotices } ) {
+		return {
+			completedNotices,
+			errorNotices,
+			inProgressNotices,
+		};
+	}
 
-	return {
-		plugin: get( sampleLog, 'plugin.name' ),
-		wp_admin_settings_page_url: get( sampleLog, 'plugin.wp_admin_settings_page_url' ),
-		numberOfPlugins,
-		site: get( sampleLog, 'site.title' ),
-		isMultiSite: get( sampleLog, 'site.is_multi_site' ),
-		numberOfSites,
-	};
-}
+	getCurrentNotices = () => {
+		const { completedNotices, errorNotices, inProgressNotices, pluginId, siteId } = this.props;
 
-class PluginNotices extends React.Component {
-	constructor( props ) {
-		super( props );
+		return {
+			errors: filterNotices( errorNotices, siteId, pluginId ),
+			inProgress: filterNotices( inProgressNotices, siteId, pluginId ),
+			completed: filterNotices( completedNotices, siteId, pluginId ),
+		};
+	};
 
-		this.state = { notices: this.refreshPluginNotices() };
+	getPluginById( pluginId ) {
+		return this.props.plugins.find( ( plugin ) => plugin.id === pluginId );
 	}
 
-	componentDidMount() {
-		PluginsLog.on( 'change', this.showNotification );
+	getCombination( translateArg ) {
+		return (
+			( translateArg.numberOfSites > 1 ? 'n sites' : '1 site' ) +
+			' ' +
+			( translateArg.numberOfPlugins > 1 ? 'n plugins' : '1 plugin' )
+		);
 	}
 
-	componentWillUnmount() {
-		PluginsLog.removeListener( 'change', this.showNotification );
-	}
+	getTranslateArg( logs, sampleLog, typeFilter ) {
+		const filteredLogs = logs.filter( ( log ) => {
+			return log.status === typeFilter ? typeFilter : sampleLog.type;
+		} );
 
-	shouldComponentUpdateNotices( currentNotices, nextNotices ) {
-		if ( currentNotices.errors && currentNotices.errors.length !== nextNotices.errors.length ) {
-			return true;
-		}
-		if (
-			currentNotices.inProgress &&
-			currentNotices.inProgress.length !== nextNotices.inProgress.length
-		) {
-			return true;
-		}
-		if (
-			currentNotices.completed &&
-			currentNotices.completed.length !== nextNotices.completed.length
-		) {
-			return true;
-		}
-		return false;
-	}
+		const numberOfSites = uniqBy( filteredLogs, 'siteId' ).length;
+		const numberOfPlugins = uniqBy( filteredLogs, 'pluginId' ).length;
+
+		const logSite = this.props.sites.find( ( site ) => parseInt( sampleLog.siteId ) === site?.ID );
+		const logPlugin = this.getPluginById( sampleLog.pluginId );
 
-	refreshPluginNotices = () => {
-		const site = this.props.selectedSite;
 		return {
-			errors: filterNotices( PluginsLog.getErrors(), site, this.props.pluginSlug ),
-			inProgress: filterNotices( PluginsLog.getInProgress(), site, this.props.pluginSlug ),
-			completed: filterNotices( PluginsLog.getCompleted(), site, this.props.pluginSlug ),
+			plugin: logPlugin?.name,
```

Comment:
`logPlugin` may not exist if we're not working with a single plugin but with multiple ones.